### ISSUE TITLE
Vector model file not found (cord19-300d.magnitude)

### ISSUE THREAD

**Issue by fomar1994** [neuml/paperai#19]
- issue moved from wrong project to here -

Hi,

I get the following error when running python -m paperai.index

raise IOError(ENOENT, "Vector model file not found", path)
FileNotFoundError: [Errno 2] Vector model file not found: 'C:\\Users\\x\\.cord19\\vectors\\cord19-300d.magnitude'

PS. I am quite new to all this; so, apologies if the mistake is on my end.

When trying to download cord19-300d.magnitude from https://www.kaggle.com/davidmezzetti/cord19-fasttext-vectors#cord19-300d.magnitude, I get the error: "Too many requests"

**Comment by davidmezzetti** [comment]
Thank you for moving the issue here. 

You may need to login to Kaggle to download the file. I have a task to make these files available on GitHub as well as Kaggle but that is still pending.

**Comment by fomar1994** [comment]
Thank you. Downloaded and unpacked in env\Lib\site-packages\paperetl\cord19

However, I still get the same error as before.
And when running python -m paperai.vectors, I get: "sqlite3.OperationalError: unable to open database file"

**Comment by davidmezzetti** [comment]
The file path should be: C:\Users\x\.cord19\vectors\cord19-300d.magnitude

I copied the path from the original message above. x should be whatever your actual local windows user name is.

**Comment by fomar1994** [comment]
Sorry about that, my mistake. It should be in the correct path now.

However, I still get the error "sqlite3.OperationalError: unable to open database file" when I run either python -m paperai.vectors or python -m paperai.index

Am I doing something wrong?

**Comment by davidmezzetti** [comment]
If you run paperai.vectors it will overwrite the vector file. My guess is that you now have an empty database. If you replace the file there, you should only need to run paperai.index.

**Comment by fomar1994** [comment]
Deleted the vector file and unpacked again to  C:\Users\x\.cord19\vectors
Only ran python -m paperai.index but still "sqlite3.OperationalError: unable to open database file"

One thing to note: the .cord19 path did not exist and I had to create it myself earlier. Is that oke?

**Comment by davidmezzetti** [comment]
Did you build an articles.sqlite database with paperetl? https://github.com/neuml/paperetl#load-pdf-articles-into-sqlite

**Comment by fomar1994** [comment]
I did but I think I may have saved it to the wrong location: C:\Users\x\Desktop\PaperAI\test-papers\sql\articles.sqlite

**Comment by fomar1994** [comment]
I have download attribute and attribute.csv to C:\Users\x\.paperai\models\attribute
design and design.csv to C:\Users\x\.paperai\models\design

When running python -m paperetl.file C:\Users\x\Desktop\PaperAI\test-papers\ ~/.paperai/models ~/.paperai/models
It starts processing the first paper but then I get:

       sock.connect(sa)
ConnectionRefusedError: [WinError 10061] No connection could be made because the target machine actively refused it

and a few more of those connection errors.

**Comment by davidmezzetti** [comment]
paperetl needs GROBID installed for PDF parsing: https://github.com/neuml/paperetl#additional-dependencies

**Comment by fomar1994** [comment]
Thanks for the help so far. I got GROBID finally working: installed and BUILD SUCCESFUL

However, when running python -m paperetl.file C:\Users\x\Desktop\PaperAI\test-papers\ ~/.paperai/models ~/.paperai/models
   
`ConnectionRefusedError: [WinError 10061] No connection could be made because the target machine actively refused it`

**Comment by davidmezzetti** [comment]
I would try GROBID locally (http://localhost:8070) and process your document. GROBID notes on their website that they don't support Windows and recommend running a docker image: https://grobid.readthedocs.io/en/latest/Troubleshooting/#windows-related-issues

**Comment by fomar1994** [comment]
Thank you for the advice. I have GROBID image running in Docker now. However, I still get connection errors.
Full error pasted here: https://pastebin.pl/view/190bc011

**Comment by davidmezzetti** [comment]
I would try to directly bring up http://localhost:8070 in a web browser. GROBID needs to run on that url. It's possible you have a local firewall blocking 8070.

**Comment by fomar1994** [comment]
It seems that I am unable to reach localhost:8070. Would you happen to know how to fix this? I will also keep looking in the meantime.

**Comment by davidmezzetti** [comment]
Currently, paperetl is hardcoded to http://localhost:8070 for GROBID. Probably should be configurable. I would work on getting your setup to have GROBID hosted on 8070.

**Comment by fomar1994** [comment]
Thanks. GROBID is now running on 8070. After the first paper starts processing, I now get this error:

```
  Resource ←[93mpunkt←[0m not found.
  Please use the NLTK Downloader to obtain the resource:

  ←[31m>>> import nltk
  >>> nltk.download('punkt')
  ←[0m
  For more information see: https://www.nltk.org/data.html

  Attempted to load ←[93mtokenizers/punkt/english.pickle←[0m'

  Searched in: ...
```

I am not sure what to do here.

**Comment by fomar1994** [comment]
I managed to use the nltk downloader to  download punkt.
But when I run
`python -m paperetl.file C:\Users\x\Desktop\PaperAI\test-papers\ C:\Users\x\.paperai\models C:\Users\x\.paperai\models
`
I get 
`PermissionError: [Errno 13] Permission denied: 'C:\\Users\\x\\.paperai\\models\\attribute'`

Also tried running as admin.

**Comment by fomar1994** [comment]
Fixed issue by moving models to desktop. After running paperetl, I got: Total articles inserted: 2

But when running python -m paperai.index, I get: sqlite3.OperationalError: no such table: sections

**Comment by davidmezzetti** [comment]
Given you moved the models out of the default location, you'll need to pass the path to the models and path to the vectors file on the command line.

The unpickle error can be safely ignored.

Thank you for continuing to document your efforts. This hopefully will help someone else someday.

**Comment by fomar1994** [comment]
I forgot to mention that I have since then moved the pre-trained study design/attribute models back to their original location. The mistake was that I had set-up the file system incorrectly (created new folders within ~/.paperai/models).

It seems to me that articles.sqlite is loaded correctly. However, running python -m paperai.index still does not work.
`sqlite3.OperationalError: no such table: sections`

Let me know if you have any ideas :)

**Comment by davidmezzetti** [comment]
When you looked at the SQLite database, did you see a table named sections?

**Comment by fomar1994** [comment]
SQLite viewer shows a dropdown menu with "articles" and "section"
I have added the test sql file here.

[articles.zip](https://github.com/neuml/paperai/files/5710536/articles.zip)

**Comment by davidmezzetti** [comment]
When you call index, are you passing the path to where the model directory is as the first argument? I know things have moved around a lot during your testing, the only way you'd get that message is if the indexing process isn't pointing to the right directory. 

```bash
python -m paperai.index <directory holding articles.sqlite> <full path to cord19 vectors file>
```

**Comment by fomar1994** [comment]
Oke this is embarrassing... I was running the command without arguments. Thanks, it seems to be running now:
```
Building new model
Iterated over 257 total rows
Iterated over 257 total rows
```
However, I don't see the model in ~/.cord19 but instead there are three files created in ~\.paperai\models.
Is that what I need?

**Comment by davidmezzetti** [comment]
That is right, since parameters were passed where to create the model directories. ~/.cord19 is just the default when not specified.

**Comment by davidmezzetti** [comment]
That is right since parameters were passed where to create the model directories. ~/.cord19 is just the default when not specified.

**Comment by fomar1994** [comment]
Could you point me in the right direction on how to use paperai.report?
`The system cannot find the path specified.`
`ERROR: loading model: ensure model is present`

I might be using the wrong arguments there.

**Comment by fomar1994** [comment]
I have been trying to follow:
https://www.kaggle.com/davidmezzetti/cord-19-analysis-with-sentence-embeddings/notebook#Testing-the-model

But not much success. So far I have loaded 19 papers into an articles.sqlite file and successfully build the model
```
Building new model
Iterated over 4712 total rows
Iterated over 4712 total rows
```

Both word embedding and sentence embedding seem to be working.
But when running the query, all I see is
```
Loading model from cord19q
Query: (my query)

Highlights

Articles
```

**Comment by fomar1994** [comment]
I just found out that all the queries I was running simply had no output. Not sure why. I tried a query that did have an output and can confirm that it's working now. However, I need to look more into what type of queries do and do not work but it seems very promising.

One question I have remaining: is the cord19 vector really optimized for covid-19 literature or should it still work fine for other biomedical literature? And is it possible to easily swap the vector?

Oh and when building a report, how do I get a similar output as shown in the demo?
